packages/synapse-react-client/src/components/CitationPopover/CitationPopover.test.tsx

```diff
@@ -0,0 +1,194 @@
+  })
+
+  it('opens popover when button is clicked and fetches citation', async () => {
+    render(<CitationPopover {...mockProps} />)
+    openPopover()
+
+    await waitFor(() => {
+      expect(screen.getByTestId('CiteAsPopover')).toBeInTheDocument()
+    })
+
+    await waitFor(() => {
+      expect(mockUseCitation).toHaveBeenCalledWith(
+        mockProps.doi,
+        'bibtex',
+        true,
+      )
+    })
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(content => content.includes('Some BibTeX Entry')),
+      ).toBeInTheDocument()
+    })
+  })
+
+  it('shows menu options when select button is clicked with bibtex as default', async () => {
+    render(<CitationPopover {...mockProps} />)
+    openPopover()
+
+    await waitFor(() => {
+      expect(screen.getByTestId('CiteAsPopover')).toBeInTheDocument()
+    })
+
+    const select = screen.getByRole('combobox')
+    await act(async () => {
+      await userEvent.click(select)
+    })
+
+    await waitFor(() => {
+      const bibtexItems = screen.getAllByText('bibtex')
+      expect(bibtexItems.length).toBeGreaterThan(0)
+      expect(screen.getByText('apa')).toBeInTheDocument()
+      expect(screen.getByText('science')).toBeInTheDocument()
+      expect(screen.getByText('nature')).toBeInTheDocument()
+      expect(screen.getByText('ieee')).toBeInTheDocument()
+    })
+  })
+
+  it('displays boilerplate text when available', async () => {
+    render(<CitationPopover {...mockProps} />)
+    openPopover()
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(content => content.includes('Some boilerplate text')),
+      ).toBeInTheDocument()
+    })
+  })
+
+  it('copies citation to clipboard', async () => {
+    render(<CitationPopover {...mockProps} />)
+    const mockWriteText = jest.fn().mockResolvedValue('copied')
+    Object.assign(navigator, {
+      clipboard: { writeText: mockWriteText },
+    })
+    openPopover()
+
+    await waitFor(() => {
+      expect(screen.getByTestId('CiteAsPopover')).toBeInTheDocument()
+    })
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(content => content.includes('Some BibTeX Entry')),
+      ).toBeInTheDocument()
+    })
+
+    const copyButton = screen.getByTestId('CopyButton')
+    await act(async () => {
+      await userEvent.click(copyButton)
+    })
+
+    await waitFor(() => {
+      expect(mockWriteText).toHaveBeenCalledWith(data)
+    })
+  })
+
+  it('downloads citation', async () => {
+    render(<CitationPopover {...mockProps} />)
+    openPopover()
+
+    await waitFor(() => {
+      expect(screen.getByTestId('CiteAsPopover')).toBeInTheDocument()
+    })
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(content => content.includes('Some BibTeX Entry')),
+      ).toBeInTheDocument()
+    })
+
+    const createElementSpy = jest
+      .spyOn(document, 'createElement')
+      .mockReturnValue(document.createElement('a'))
```

Comment:
Do you need this `mockReturnValue` if you're just going to call the method itself?

```suggestion
    const createElementSpy = jest
      .spyOn(document, 'createElement')
```